torchmetrics/metric.py

```diff
@@ -242,6 +270,35 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
         return self._forward_cache
 
+    def _reduce_states(self, incoming_state: Dict[str, Any]) -> None:
+        """Adds an incoming metric state to the current state of the metric.
+
+        Args:
+            incoming_state: a dict containing a metric state similar metric itself
+        """
+        for attr in self._defaults.keys():
+            local_state = getattr(self, attr)
+            global_state = incoming_state[attr]
+            reduce_fn = self._reductions[attr]
+            if reduce_fn == dim_zero_sum:
+                reduced = global_state + local_state
+            elif reduce_fn == dim_zero_mean:
+                reduced = (self._update_count * global_state + local_state) / (self._update_count + 1)
```

Comment:
shouldn't this be `((self._update_count -1) * global_state + local_state) / self._update_count` as the count was already updated by the preceeding update call for local state?

Reply:
yes, you are right, fixing :)